### DISCUSSION TITLE
QUIC & HTTP/3 support

### DISCUSSION THREAD

**Discussion by adworacz** [DNSCrypt/dnscrypt-proxy#1759] [Ideas]
With Adguard supporting [DNS over QUIC](https://adguard.com/en/blog/dns-over-quic.html) and NextDNS adding QUIC/HTTP3 to their [client](https://github.com/nextdns/nextdns/issues/390#issuecomment-801513998) and [servers](https://github.com/nextdns/nextdns/issues/390#issuecomment-801573265) (overall [support doc](https://help.nextdns.io/t/x2hmvas/what-is-dns-over-tls-dot-dns-over-quic-doq-and-dns-over-https-doh-doh3)), has there been any thoughts to adding similar support to `dnscrypt-proxy`?

HTTP3 would essentially be an extension of the existing DoH functionality I'd imagine. Not sure about opinions on DoQ (QUIC) though.

**Comment by jedisct1** [comment]
Once https://github.com/golang/go/issues/32204 is closed, I guess dnscrypt-proxy will immediately get support for HTTP/3.

But DNSCrypt already provides encrypted DNS over UDP, with less overhead and less attack surface than QUIC.

**Reply by adworacz** [reply]
Got it. Thanks for the details re: HTTP/3.

And I figured DNSCrypt would be more ideal than QUIC. Alas, NextDNS (my current provider of choice) still doesn't support DNSCrypt yet.
